<a href="https://colab.research.google.com/github/hag86/BTE-320/blob/main/First_in_class_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
a==b
a!=b

a<b
a<=b
a>b
a>=b


Hello


In [ ]:
3+2

5

In [ ]:
3.0+2.0

5.0

In [ ]:
3+2.0

5.0

In [ ]:
3-1

2

In [ ]:
2-4

-2

In [ ]:
2*3

In [ ]:
4/2

2.0

In [ ]:
5/2

2.5

In [ ]:
4//2

2

In [ ]:
5//2

2

In [ ]:
2**3

8

In [ ]:
pow(2,3)


8

In [ ]:
3%2

1

In [ ]:
6%3

0

In [ ]:
3%5

3

In [ ]:
2%3

2

In [ ]:
a=True
b=False
c=True
d=True
e=False

In [ ]:
# Inputs - assign data to variables
pi=3.14
radius=3

# Processes - calculate the area and assign it to a variable
area=pi*(radius**2)

#outputs - display area on the screen
print(area)

28.26


In [ ]:
radius = 3
pi = 3.14

radius, pi = 3, 3.14

In [ ]:
x,y=2,3
print(x,y)

2 3


In [ ]:
x,y=y,x
print(x,y)


3 2


In [ ]:
print(2, 3)

2 3


In [ ]:
print(x,y)

3 2


print(obj(s), sep=" ", end="\n", file=..., flush=...)

In [ ]:
print("Hello", "there", end="\n\n\n")
print("Hi")

Hello there


Hi


In [ ]:
print("Hello", "there", sep="---", end=" ")
print("Hi")

Hello---there Hi


'''python
"Hello" + "---" + "there" + " "
"Hello---there "

In [ ]:
print("This is\nin two lines")

This is
in two lines


In [ ]:
month="August"
day=27
year=2025

print("Today is", month, day, ",", year)

Today is August 27 , 2025


In [ ]:
"Today is {} {}, {}".format(month, day, year)

'Today is August 27, 2025'

In [ ]:
s = "Today is {} {}, {}".format(month, day, year)

In [ ]:
print (s)

Today is August 27, 2025


In [ ]:
month="August"
day=27
year=2025

print("Today is", month, day, ",", year)
print("Today is {} {}, {}".format(month, day, year))
print(f"Today is {month} {day}, {year}")

Today is August 27 , 2025
Today is August 27, 2025
Today is August 27, 2025


In [ ]:
month= input("Enter month: ")
print(month, type(month))

Enter month: August
August <class 'str'>


In [ ]:
day = input("Enter day: ")
day = float(day)
print(day, type(day))

Enter day: 3
3.0 <class 'float'>


# Inventory management

In [ ]:
import math

#Inputs
demand = int(input("Enter demand (units/year): "))
reorder = float(input("Enter reorder cost ($/order): "))
holding = float(input("Enter holding cost ($/unit/year): "))

# Process
eoq = (2*demand*reorder/holding) ** (1/2)

# Output
print(f"EOQ: {eoq} units")

Enter demand (units/year): 1000
Enter reorder cost ($/order): 50
Enter holding cost ($/unit/year): 10
EOQ: 100.0 units


# 9/8 Class Notes


In [ ]:
x=5

if x>3:
  print("x is greater than 3.")

  print("Press any key to continue...")

x is greater than 3.
Press any key to continue...


In [ ]:
grade = 90

if grade >= 90:
  print("A")
else:
  print("A-")

print("Goodbye...")

A
Goodbye...


In [ ]:
grade = 90

if grade >= 97:
  print ("A+")
elif grade >= 93:
  print("A")
elif grade >= 90:
  print("A-")
elif grade >= 88:
  print("B+")
else:
  print("B")

A-


# 9/10 class notes

In [ ]:
x= int(input('Enter a number:'))

if x % 2 == 0:
  if x % 3 == 0:
    print(f"{x} is divisible by both. ")
  else:
    print(f"{x} is divisible by 2 but not by 3.")
elif x % 3 == 0:
  print(f"{x} is divisble by 3 but not by 2.")
else:
  print(f"{x} is divisible by none of them.")

Enter a number:8
8 is divisible by 2 but not by 3.


In [ ]:
x = int(input('Enter a number:'))

if x % 2 == 0 and x % 3 == 0:
  print (f"{x} is divisible by both.")
elif x % 2 == 0 and x % 3 != 0:
  print(f"{x} is divisible by 2 only.")
elif x % 3 == 0 and x % 2 !=0:
  print(f"{x} is divisible by 3 only.")
else:
  print(f"{x} is divisible by none of them.")

Enter a number:9
9 is divisible by 3 only.


In [ ]:
x = int(input('Enter a number:'))

print('Even' if x % 2 == 0 else 'Odd')


# Numerical calculator


# Design a simple numerical calculator in Python. The user needs to provide three inputs: two numbers and the mathematical operator of their choice. The program should print the output on the screen.


In [ ]:
n1 = float(input("Enter number 1: "))
n2 = float(input("Enter number 2:"))
op = input("Enter operator (+,-,*,/,//,%,**)")

if op == '+':
  res = n1 + n2
elif op == "-":
  res = n1 - n2
elif op == "*":
  res = n1 * n2
elif op == "/":
  res = n1 / n2
elif op == "//":
  res = n1 // n2
elif op == "**":
  res = n1 ** n2
elif op == "%":
  res = n1 % n2
else:
  res = None
  print(f"Invalid operator")

print(f"result = {res}")

Enter number 1: 5
Enter number 2:2
Enter operator (+,-,*,/,//,%,**)**
result = 25.0


# 9/15

In [ ]:
from ColabTurtlePlus.Turtle import *
# setting up the screen commands
clearscreen ()
reset = True
clear()

#initialize drawing process
color('black, 'green')
jumpto(0,0)

for _ in range(4):
  forward(100)
  left(90)


SyntaxError: unterminated string literal (detected at line 8) (ipython-input-3287394597.py, line 8)

In [ ]:
for num in range(0,11):
  print num

SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)? (ipython-input-3568970596.py, line 2)

# 9/17

A parking garage charges $5 plus $2.50 for eacg hour parked.

Write a program that prints the number of hours parked and the corresponding fee side by side, from 1 to 8 hours.

Example output:
1 7.5
2 10.0
3 12.5
4 15.0
5 17.50
6 20.0
7 22.5
8 25.0

Bonus: The minimum fee is $10 and the maximum fee is $20

In [ ]:
for hour in range (1,9):
  fee = 5 + 2.5 * hour

  if fee < 10:
    fee = 10.0
  elif fee > 20:
    fee = 20.0
  print(hour, fee)




1 10.0
2 10.0
3 12.5
4 15.0
5 17.5
6 20.0
7 20.0
8 20.0


In [ ]:
for number in range(1,4):
  print(number, number ** 2)

1 1
2 4
3 9


In [ ]:
number = 1
while number < 4:
  print (number, number **2)
  number = number + 1

1 1
2 4
3 9


In [ ]:
number = float(input("Give a number: "))
while number != 0:
  print(number)
  number = float(input("Give another number: "))

Give a number: 5
5.0
Give another number: -9
-9.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 5
5.0
Give another number: 


ValueError: could not convert string to float: ''

In [ ]:
for number in range(10):
   if number % 2 != 0:
    continue

    print(number)


In [ ]:
for i in range (100000000000000):
  if i == 3:
    break
  print (i)

0
1
2


# 9/22

A movie theater manager is deciding how much to spend on advertisising for an upcoming movie. Each ticket is $10. Without any advertising, 20 people are forecasted to attend. Additional people may attend based on advertising spending, estimated to be:
  Additional attendees = 2 x advertisingDollars^1/2
In addition to advertising costs, there are fixed costs to run the theater of $200

Write a program to generate a table for different advertising amounts and profits. Example output(may differ slightly from your results):

In [ ]:
print("Ad ($) Profit ($)")
for ad in range(0,201,25):
  revenue = (20 + 2 * ad ** 0.5) * 10
  costs = 200 + ad
  profit = revenue - costs
  print(f"{ad}, \t {round(profit)}")

Ad ($) Profit ($)
0, 	 0
25, 	 75
50, 	 91
75, 	 98
100, 	 100
125, 	 99
150, 	 95
175, 	 90
200, 	 83


ch.3 / lecture 3

In [ ]:
phrase = "This is a string"

In [ ]:
len(phrase)

16

In [ ]:
phrase[5]

'i'

In [ ]:
s = phrase[:7]
print(phrase)

This is a string


In [ ]:
phrase[0:4]

'This'

In [ ]:
phrase[:4]

'This'

In [ ]:
phrase[10:16]

'string'

In [ ]:
phrase[:]

'This is a string'

In [ ]:
phrase[:] == phrase[0:16] == phrase[0:len(phrase)] == phrase

True

In [ ]:
phrase[3] = 'b'

TypeError: 'str' object does not support item assignment

In [ ]:
'Hello' + 'there'

'Hellothere'

In [ ]:
3 * 'a'

'aaa'

In [ ]:
3 * 'abc'

'abcabcabc'

In [ ]:
s = ""
len(s)

0

In [ ]:
L = []
len(L)

0

In [ ]:
L = [1,2,3,4,5]

In [ ]:
L[3]

4

In [ ]:
L[1:4]

[2, 3, 4]

In [ ]:
[1,2,3] + ['one','two', 'three']

[1, 2, 3, 'one', 'two', 'three']

In [ ]:
3 * [1,2]

[1, 2, 1, 2, 1, 2]

In [ ]:
s = 'abc123'

In [ ]:
L[2] = 6
print(L)

[1, 2, 6, 4, 5]


In [ ]:
L = [1, 'two', 3.14, [4,5,6], print, len]
len(L)

6

In [ ]:
".....{}...{}..." . format(a,b)

In [ ]:
print(L)

[1, 'two', 3.14, [4, 5, 6], <built-in function print>, <built-in function len>]


In [ ]:
L.append(3.0)

In [ ]:
print(L)

[1, 'two', 3.14, [4, 5, 6], <built-in function print>, <built-in function len>, 3.0, 3.0]


In [ ]:
L.append('abc')

In [ ]:
print(L)

[1, 'two', 3.14, [4, 5, 6], <built-in function print>, <built-in function len>, 3.0, 3.0, 3.0, 'abc']


In [ ]:
L.extend('abc')
print(L)

[1, 'two', 3.14, [4, 5, 6], <built-in function print>, <built-in function len>, 3.0, 3.0, 3.0, 'abc', 'a', 'b', 'c']


In [ ]:
L.pop(5)
print(L)

[1, 'two', 3.14, [4, 5, 6], <built-in function print>, 3.0, 3.0, 3.0, 'abc', 'a', 'b', 'c']


In [ ]:
L = [5,8,0,1,6,2]
L.sort(reverse=True)
print(L)

[8, 6, 5, 2, 1, 0]


In [ ]:
Ls = ['bob', 'alice', 'eve']
Ls.sort()
print(Ls)

['alice', 'bob', 'eve']


In [ ]:
L = [5,8, 'two', 'three', 1]
L.sort()

TypeError: '<' not supported between instances of 'str' and 'int'

Append items to list iteratively (while loop)
- starting with an empty list, desgin a program in python that populates the list by repeatedly asking the user for nurmerical inputs. The process stops when the user gives 0. Finally, the list should be displayed in descending order.

In [ ]:
numbers = []
num = float(input("Give a number: "))

while num != 0:
  numbers.append(num)
  num = float(input("Give another number: "))

numbers.sort(reverse=True)
print(numbers)


Give a number: 5
Give another number: 8
Give another number: 6
Give another number: 2
Give another number: 0
[8.0, 6.0, 5.0, 2.0]


# 9/29 notes

In [ ]:
L = [10.0,9.9,10.1,10.05]

In [ ]:
max(L)

10.1

In [ ]:
sum(L)

40.05

In [ ]:
sum(L)/ len(L)

10.0125

In [ ]:
t = (1,2,3)
len(t)

3

In [ ]:
t[1]

2

In [ ]:
t[1:3]

(2, 3)

In [ ]:
(1,2) + ('a','b')

(1, 2, 'a', 'b')

In [ ]:
2 * (1,2)

(1, 2, 1, 2)

In [ ]:
t[1] = 3

TypeError: 'tuple' object does not support item assignment

In [ ]:
type([1])

list

In [ ]:
type((1))

int

In [ ]:
[1] == [1,]

True

In [ ]:
grades = {
    'Bob' : 95,
    'Alice': 98,
    'Eve': 92,
}

len(grades)

3

In [ ]:
grades['Alice']

98

In [ ]:
grades['Eve'] = 96
print(grades)

{'Bob': 95, 'Alice': 98, 'Eve': 96}


In [ ]:
grades['Steve'] = 91
print(grades)

{'Bob': 95, 'Alice': 98, 'Eve': 96, 'Steve': 91}


In [ ]:
grades.keys()

dict_keys(['Bob', 'Alice', 'Eve', 'Steve'])

In [ ]:
grades.values()

dict_values([95, 98, 96, 91])

In [ ]:
grades.items()

dict_items([('Bob', 95), ('Alice', 98), ('Eve', 96), ('Steve', 91)])

In [ ]:
for key in grades.keys():
  print(grades[key])

95
98
96
91


In [ ]:
sum(grades.values())/ len(grades.values())

95.0

In [ ]:
('Dave' in grades.keys()) == ('Dave' in grades)

True

In [ ]:
for key, value in grades.items():
  print(key, value)

Bob 95
Alice 98
Eve 96
Steve 91


In [ ]:
clients = {
    'johndoe@aol.com' :{
        'First': 'John',
        'Last': 'Doe',
        'Phone': '+1 (555) 123-4567',
    }
}

print(clients)

{'johndoe@aol.com': {'First': 'John', 'Last': 'Doe', 'Phone': '+1 (555) 123-4567'}}


Using python in colab: The classic hand game. Write a program that pays the fame against the computer for as long as the user wants. Start by allowing the user to choose whether they want to play using the appropriate input. As long as the use chooses 'yes', the game continues. Add realism by allowing the ocmputer to randomly choose its action. You can do thisusing the random library: import random, computer = random.choice(['rock', 'paper', 'scissors']

In [ ]:
import random

print("Welcome to Rock-Paper-Scissors!")

# Main game loop
while True:
    play = input("Do you want to play? (yes/no): ").strip().lower()
    if play != "yes":
        print("Thanks for playing! Goodbye.")
        break

    # Get user's choice
    user = input("Choose rock, paper, or scissors: ").strip().lower()
    if user not in ["rock", "paper", "scissors"]:
        print("Invalid choice. Please try again.")
        continue

    # Computer's random choice
    computer = random.choice(["rock", "paper", "scissors"])
    print(f"Computer chose: {computer}")

    # Decide winner
    if user == computer:
        print("It's a tie!")
    elif (user == "rock" and computer == "scissors") or \
         (user == "scissors" and computer == "paper") or \
         (user == "paper" and computer == "rock"):
        print("You win!")
    else:
        print("You lose!")


Welcome to Rock-Paper-Scissors!
Do you want to play? (yes/no): yes
Choose rock, paper, or scissors: paper
Computer chose: paper
It's a tie!
Do you want to play? (yes/no): yes
Choose rock, paper, or scissors: scissors
Computer chose: rock
You lose!
Do you want to play? (yes/no): yes
Choose rock, paper, or scissors: scissors
Computer chose: paper
You win!
Do you want to play? (yes/no): no
Thanks for playing! Goodbye.


```python
user_choice == 'rock' and pc_choice

In [ ]:
import random

while True:

  play = input('Do you want to play a game (yes/no)? ').strip().lower()

  if play == 'no': # == play != 'yes'
    break

  # Rock, paper, scissors starts here...
  user_choice = input('Choose between rock/paper/scissors: ')
  pc_choice = random.choice(['rock','paper', 'scissors'])
  print(f'User: {user_choice}, Computer: {pc_choice}')

  if user_choice == pc_choice:
    print('Tie')
  elif (
      (user_choice == 'rock' and pc_choice == 'scissors') or
      (user_choice =='scissors' and pc_choice == 'paper') or
      (user_choice =='paper' and pc_choice == 'rock')
  ):
    print('User Wins!')
  else:
    print('Computer wins')

Do you want to play a game (yes/no)? yes
Choose between rock/paper/scissors: rock
User: rock, Computer: paper
Computer wins
Do you want to play a game (yes/no)? yes
Choose between rock/paper/scissors: paper
User: paper, Computer: scissors
Computer wins
Do you want to play a game (yes/no)? no


In [ ]:
s1 = set([1,2,2,3,4,5,5,6,7,8])
print(s1)

{1, 2, 3, 4, 5, 6, 7, 8}


In [ ]:
len(s1)

8

In [ ]:
s2 = {5,2,7,8,9,11}
print(s2)

{2, 5, 7, 8, 9, 11}


In [ ]:
s1.intersection(s2)

{2, 5, 7, 8}

In [ ]:
s2.intersection(s1)

{2, 5, 7, 8}

In [ ]:
s1 & s2

{2, 5, 7, 8}

In [ ]:
s1.union(s2)

{1, 2, 3, 4, 5, 6, 7, 8, 9, 11}

In [ ]:
s1.difference(s2)

{1, 3, 4, 6}

In [ ]:
s2.difference(s1)

{9, 11}

In [ ]:
s1 - s2

{1, 3, 4, 6}

In [ ]:
s1.symetric_difference(s2)

AttributeError: 'set' object has no attribute 'symetric_difference'

Product prices (dictionaries)
A company sells two products, their prices are:
P-01 $9.95
P-02 $10.50
Write a program that sets up a dictionary with product IDs as keys and prices as their values. Prompt the user to enter a product ID, then display that product's price

sample run:
Enter product ID: P-01
P-01 9.95
For your program with a dictionary of product IDs and their prices, refine it so that if the user enters a product ID not in the dictionary it display a message
Sample run:
Enter product ID: X-01
X-01 not found

In [ ]:
products = {
    'P-01': 9.95,
    'P-02': 10.50
}

id = input('Enter product ID: ')

if id in products:
  print(f'ID: {id}, Price: {products[id]}$')
else:
  print(f'{id} not found')


Enter product ID: P-02
ID: P-02, Price: 10.5$
